router.go

```diff
@@ -272,12 +272,12 @@ func (n *node) check405() HandlerFunc {
 			return methodNotAllowedHandler
 		}
 	}
-	return notFoundHandler
+	return methodNotAllowedHandler
```

Comment:
It's also worth pointing out this line here. I changed this to `methodNotAllowedHandler`, as I believe that using `notFoundHandler` was incorrect here.